Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -599,23 +580,36 @@ void USpatialActorChannel::RegisterEntityId(const FEntityId& ActorEntityId)
 	}
 }
 
-void USpatialActorChannel::OnCreateEntityResponse(const worker::CreateEntityResponseOp& Op)
+void USpatialActorChannel::OnReserveEntityIdResponse(const Worker_ReserveEntityIdResponseOp& Op)
+{
+	if (Op.status_code != WORKER_STATUS_CODE_SUCCESS)
+	{
+		UE_LOG(LogSpatialGDKActorChannel, Error, TEXT("Failed to reserve entity id. Reason: %s"), UTF8_TO_TCHAR(Op.message));
```

Comment:
One way we can improve this is to implement an "EntityIdAllocator" which fetches batches of EntityIds using `ReserveEntityIds` on demand when needed. The idea is that the actor channel will "request" an entity ID and the "EntityIdAllocator" can (eventually) return an entity ID for use. If we allocate many entity IDs at once, then most of the time this will take a negligible amount of time (rather than having to wait for a round trip to the runtime).

Reply:
From a conversation I saw elsewhere, apparently reserved entity ids expire after a certain period of time? If there was a way to stop this from happening I think it makes sense to allocate in bunches.